spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoTemplate.java

```diff
@@ -656,6 +656,17 @@ public boolean exists(Query query, Class<?> entityClass, String collectionName)
 		return doFindOne(collectionName, new BasicDBObject(idKey, id), null, entityClass);
 	}
 
+	public <T> List<T> distinct(Query query, String field, String collectionName) {
+		DBCollection collection = this.getCollection(collectionName);
+		List list = collection.distinct(field, query.getQueryObject());
```

Comment:
Your change does not use our mapping layer so results will be returned as `Document` instead of mapped entities.